sdk_lib/sdk_entry.sh

```diff
@@ -15,24 +15,31 @@ chown -R sdk:sdk /home/sdk
 (
     source /etc/lsb-release # SDK version in DISTRIB_RELEASE
     source /mnt/host/source/.repo/manifests/version.txt # OS image version in FLATCAR_VERSION_ID
+    version="${FLATCAR_VERSION_ID}"
 
-    if [ "${FLATCAR_VERSION_ID}" != "${DISTRIB_RELEASE}" ] ; then
+    # If this is a nightly build tag we can use pre-built binaries directly from the
+    #  build cache.
+    if [[ "${FLATCAR_BUILD_ID}" =~ ^nightly-.*$ ]] ; then
+        version="${FLATCAR_VERSION_ID}+${FLATCAR_BUILD_ID}"
+    fi
+
+    if [ "${version}" != "${DISTRIB_RELEASE}" ] ; then
         for target in amd64-usr arm64-usr; do
             if [ ! -d "/build/$target" ] ; then
                 continue
             fi
             if [ -f "/build/$target/etc/target-version.txt" ] ; then
                 source "/build/$target/etc/target-version.txt"
-                if [ "${TARGET_FLATCAR_VERSION_ID}" = "${FLATCAR_VERSION_ID}" ] ; then
+                if [ "${TARGET_FLATCAR_VERSION}" = "${version}" ] ; then
```

Comment:
I don't understand where the change from `TARGET_FLATCAR_VERSION_ID` to `TARGET_FLATCAR_VERSION` comes from.

Reply:
Oh, I forgot to update the other occurrence of the variable, in line 42.

Reply:
Changed the name from `TARGET_FLATCAR_VERSION_ID` to `TARGET_FLATCAR_VERSION` because with `version` being `"${FLATCAR_VERSION_ID}+${FLATCAR_BUILD_ID}"` in case of a nightly build there's no 1:1 relation to `FLATCAR_VERSION_ID` anymore (at least not for nightlies).